adapter/internal/discovery/xds/server.go

```diff
@@ -425,22 +439,82 @@ func GenerateEnvoyResoucesForLabel(label string) ([]types.Resource, []types.Reso
 		vhostToRouteArrayMap[systemHost] = append(vhostToRouteArrayMap[systemHost], readynessEndpoint)
 	}
 
-	listenerArray, listenerFound := envoyListenerConfigMap[label]
-	routesConfig, routesConfigFound := envoyRouteConfigMap[label]
-	if !listenerFound && !routesConfigFound {
-		listenerArray, routesConfig = oasParser.GetProductionListenerAndRouteConfig(vhostToRouteArrayMap)
-		envoyListenerConfigMap[label] = listenerArray
-		envoyRouteConfigMap[label] = routesConfig
-	} else {
-		// If the routesConfig exists, the listener exists too
-		oasParser.UpdateRoutesConfig(routesConfig, vhostToRouteArrayMap)
```

Comment:
you have removed oasParser.UpdateRoutesConfig call from api update path. is this intentional?

Reply:
yes, both flows handles from add path now